bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/internal/items/ItemUpdater.java

```diff
@@ -67,7 +79,14 @@ protected void receiveUpdate(ItemStateEvent updateEvent) {
                     }
                 }
                 if (isAccepted) {
+                    State oldState = item.getState();
                     item.setState(newState);
+                    if (oldState != null && !oldState.equals(newState)) {
```

Comment:
If a state must not be <code>null</code> but <code>UnDefType.NULL</code> it seems okay, to remove the null check (at least for me).